circle.yml

```diff
@@ -0,0 +1,21 @@
+machine:
+  xcode:
+    version: "8.0"
+
+dependencies:
+  pre:
+    - brew update 1> /dev/null 2> /dev/null
+    - brew outdated carthage || brew upgrade carthage
+
+checkout:
+  post:
+    - git submodule update --init --recursive
+test:
+  override:
+    - ./scripts/macOS
```

Comment:
#nth since the build scripts are all so similar, they could be refactored to be the same script four times with passed variables for project, scheme, sdk, and destination.